core/src/main/java/org/apache/carbondata/core/util/TrashUtil.java

```diff
@@ -0,0 +1,162 @@
+  public static void copyFileToTrashFolder(String filePathToCopy,
+      String trashFolderWithTimestamp) throws IOException {
+    CarbonFile carbonFileToCopy = FileFactory.getCarbonFile(filePathToCopy);
+    try {
+      if (carbonFileToCopy.exists()) {
+        if (!FileFactory.isFileExist(trashFolderWithTimestamp)) {
+          FileFactory.mkdirs(trashFolderWithTimestamp);
+        }
+        if (!FileFactory.isFileExist(trashFolderWithTimestamp + CarbonCommonConstants
+            .FILE_SEPARATOR + carbonFileToCopy.getName())) {
+          copyToTrashFolder(filePathToCopy, trashFolderWithTimestamp + CarbonCommonConstants
+              .FILE_SEPARATOR + carbonFileToCopy.getName());
+        }
+      }
+    } catch (IOException e) {
+      LOGGER.error("Error while creating trash folder or copying data to the trash folder", e);
+      throw e;
+    }
+  }
+
+  /**
+   * The below method copies the complete segment folder to the trash folder. Here, the data files
+   * in segment are listed and copied one by one to the trash folder.
+   *
+   * @param segmentPath the folder which are to be moved to the trash folder
+   * @param trashFolderWithTimestamp trashfolderpath with complete timestamp and segment number
+   * @return
+   */
+  public static void copySegmentToTrash(CarbonFile segmentPath,
+      String trashFolderWithTimestamp) throws IOException {
+    try {
+      List<CarbonFile> dataFiles = FileFactory.getFolderList(segmentPath.getAbsolutePath());
+      for (CarbonFile carbonFile : dataFiles) {
+        copyFileToTrashFolder(carbonFile.getAbsolutePath(), trashFolderWithTimestamp);
+      }
+      LOGGER.info("Segment: " + segmentPath.getAbsolutePath() + " has been copied to" +
+          " the trash folder successfully");
+    } catch (IOException e) {
+      LOGGER.error("Error while getting folder list for the segment", e);
+      throw e;
+    }
+  }
+
+  /**
+   * The below method deletes timestamp subdirectories in the trash folder which have expired as
+   * per the user defined retention time
+   */
+  public static void deleteExpiredDataFromTrash(String tablePath) {
+    long retentionMilliSeconds = CarbonProperties.getInstance().getTrashFolderRetentionTime();
+    String trashPath = CarbonTablePath.getTrashFolderPath(tablePath);
+    // Deleting the timestamp based subdirectories in the trashfolder by the given timestamp.
+    try {
+      if (FileFactory.isFileExist(trashPath)) {
+        List<CarbonFile> timestampFolderList = FileFactory.getFolderList(trashPath);
+        long currentTime = System.currentTimeMillis();
+        for (CarbonFile timestampFolder : timestampFolderList) {
+          long trashFolderTimestampSubFolder = Long.parseLong(timestampFolder.getName());
+          // If the timeStamp at which the timeStamp subdirectory has expired as per the user
+          // defined value, delete the complete timeStamp subdirectory
+          if (trashFolderTimestampSubFolder < currentTime - retentionMilliSeconds) {
```

Comment:
current - trashTime > retention time is more readable

Reply:
done